commit fcd2d784b8edc21f950e520c1093869666408f58
Author: iosub <iosub@gabiges.com>
Date:   Thu Nov 13 19:26:08 2025 -0500

    Backup snapshot 2025-11-12

diff --git a/llama/llama.cpp/include/llama.h b/llama/llama.cpp/include/llama.h
index 8abee5d8..8fcc0f83 100644
--- a/llama/llama.cpp/include/llama.h
+++ b/llama/llama.cpp/include/llama.h
@@ -491,6 +491,7 @@ extern "C" {
     LLAMA_API int32_t llama_model_n_head     (const struct llama_model * model);
     LLAMA_API int32_t llama_model_n_head_kv  (const struct llama_model * model);
     LLAMA_API int32_t llama_model_n_swa      (const struct llama_model * model);
+    LLAMA_API uint32_t llama_model_n_pos_per_embd(const struct llama_model * model);
 
     // Get the model's RoPE frequency scaling factor
     LLAMA_API float llama_model_rope_freq_scale_train(const struct llama_model * model);
