commit fcd2d784b8edc21f950e520c1093869666408f58
Author: iosub <iosub@gabiges.com>
Date:   Thu Nov 13 19:26:08 2025 -0500

    Backup snapshot 2025-11-12

diff --git a/llama/llama.cpp/src/llama-model.cpp b/llama/llama.cpp/src/llama-model.cpp
index 84d7183d..666aeafa 100644
--- a/llama/llama.cpp/src/llama-model.cpp
+++ b/llama/llama.cpp/src/llama-model.cpp
@@ -7434,6 +7434,10 @@ int32_t llama_model_n_swa(const llama_model * model) {
     return model->hparams.n_swa;
 }
 
+uint32_t llama_model_n_pos_per_embd(const struct llama_model * model) {
+    return model->hparams.n_pos_per_embd();
+}
+
 uint32_t llama_model_n_cls_out(const struct llama_model * model) {
     return model->hparams.n_cls_out;
 }
